packages/analytics/client/src/analytics_client/analytics_client.test.ts

```diff
@@ -311,54 +314,63 @@ describe.skip('AnalyticsClient', () => {
       expect(optIn).toHaveBeenCalledWith(true);
     });
 
-    test('Spreads the context updates to the shipper (only after opt-in)', async () => {
-      const extendContextMock = jest.fn();
-      analyticsClient.registerShipper(MockedShipper, { extendContextMock });
-      expect(extendContextMock).toHaveBeenCalledTimes(0); // Not until we have opt-in
-      analyticsClient.optIn({ global: { enabled: true } });
-      await delay(10);
-      expect(extendContextMock).toHaveBeenCalledWith({}); // The initial context
-
-      const context$ = new Subject<{ a_field: boolean }>();
-      analyticsClient.registerContextProvider({
-        name: 'contextProviderA',
-        schema: {
-          a_field: {
-            type: 'boolean',
-            _meta: {
-              description: 'a_field description',
+    test(
+      'Spreads the context updates to the shipper (only after opt-in)',
+      fakeSchedulers(async (advance) => {
```

Comment:
NIT: No need for the async keyword anymore (also valid for a few more tests below)